### PR TITLE
Adopt ISO 8601 format with calendar date, hour, minute, and second

### LINKED ISSUE #1487: Adopt ISO 8601 format with calendar date, hour, minute, and second
Current dates and times used by different components of Cruise Control are inconsistent and confusing.
* In some cases, dates use week year (`YYYY`) rather than year (`yyyy`), which would generate incorrect dates during end/beginning of years.
* Date formatting is still based on legacy classes such as java.util.Date
* Printed dates inconsistently specify the time zone and may even rely implicitly on the JVM’s current default time zone.

We should definitely switch to `ISO 8601` and use it consistently.
-- https://xkcd.com/1179/

### PR DESCRIPTION
**Pull request by efeg**
This PR resolves #1487.

### CODE REVIEW

File: cruise-control-core/src/test/java/com/linkedin/cruisecontrol/CruiseControlUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    for (Map.Entry<Long, String> entry : RESPONSE_BY_TIME_MS.entrySet()) {
+      assertEquals(entry.getValue(), CruiseControlUtils.utcDateFor(entry.getKey()));
+    }
+
+    Instant currentUtcDateFor = Instant.parse(CruiseControlUtils.utcDateFor(Instant.now().toEpochMilli()));
+    Instant currentUtcDate = Instant.parse(CruiseControlUtils.currentUtcDate());
+    long difference = currentUtcDate.minus(currentUtcDateFor.getEpochSecond(), ChronoUnit.SECONDS).getEpochSecond();
+
+    // Maximum difference accounts for a potential GC pause -- ideally (1) currentUtcDateFor and (2) currentUtcDate should be the same.
+    assertTrue(difference >= 0 && difference < MAX_ESTIMATED_PAUSE);
```
Comment (zornhsu) [inline cruise-control-core/src/test/java/com/linkedin/cruisecontrol/CruiseControlUtilsTest.java:40]: nit: difference >= `0L`?

Comment (efeg) [reply cruise-control-core/src/test/java/com/linkedin/cruisecontrol/CruiseControlUtilsTest.java:40]: That should be fine in Java. The `int` `0` will be implicitly converted to a long `0L`.

Comment (zornhsu) [reply cruise-control-core/src/test/java/com/linkedin/cruisecontrol/CruiseControlUtilsTest.java:40]: Sounds good

File: cruise-control-core/src/main/java/com/linkedin/cruisecontrol/CruiseControlUtils.java [hunk 2]
```diff
@@ -28,14 +28,18 @@ public static void ensureValidString(String fieldName, String toCheck) {
...
-  public static String toPrettyTime(long time) {
-    SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd_hh:mm:ss");
-    return sdf.format(new Date(time));
+  public static String currentUtcDate() {
+    return Instant.now().truncatedTo(ChronoUnit.SECONDS).toString();
   }
 
-
+  /**
+   * @see <a href="https://xkcd.com/1179/">https://xkcd.com/1179/</a>
```
Comment (zornhsu) [inline cruise-control-core/src/main/java/com/linkedin/cruisecontrol/CruiseControlUtils.java:38]: Seeing this image making into real commit is amusing. : )

**Review by zornhsu — APPROVED** [review]

**Comment by zornhsu** [comment]
LGTM. Thanks for unifying all the formats, Efe!

**Pull request merged**